test/hotspot/jtreg/compiler/vectorization/TestFloatConversionsVector.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8294588
+ * @summary Auto-vectorize Float.floatToFloat16, Float.float16ToFloat API's
+ * @requires vm.compiler2.enabled
+ * @requires vm.cpu.features ~= ".*avx.*"
+ * @requires os.simpleArch == "x64"
+ * @library /test/lib /
+ * @run driver compiler.vectorization.TestFloatConversionsVector
+ */
+
+package compiler.vectorization;
+
+import compiler.lib.ir_framework.*;
+
+public class TestFloatConversionsVector {
+  private static final int ARRLEN = 1024;
+  private static final int ITERS  = 11000;
+  private static float  [] finp;
+  private static short  [] sout;
+  private static short  [] sinp;
+  private static float  [] fout;
+
+  public static void main(String args[]) {
+      TestFramework.runWithFlags("-XX:-TieredCompilation",
+                                 "-XX:CompileThresholdScaling=0.3");
+      System.out.println("PASSED");
+  }
+
+  @Test
+  @IR(counts = {IRNode.VECTOR_CAST_F2H, "> 0"}, applyIfCPUFeature = {"avx512f", "true"})
```

Comment:
You can add "FC16" also in the feature list and use applyIfCPUFeaturesOr

Reply:
This is good suggestion.